### ISSUE TITLE
feature: developing plugins locally before publishing to hub

### ISSUE THREAD

**Issue by aphethean1** [meltano/meltano#6615]
### Feature scope

Configuration (settings parsing, validation, etc.)

### Description

**As a user, I want to develop plugins with dependencies before publishing these plugins to the HUB**

### Background

We use a `file` bundle type plugin to distribute analytics:  `analyze-google-analytics`

We use a `file` bundle type plugin to distribute models:  `dbt-google-analytics`

We use an `extractor` type plugin to import data:  `tap-google-analytics`

To work correctly:  `analyze` requires `models` requires `tap`

## problem

The developer wants to create `analyze` and `models` at the same time, using an existing `extractor`

But `meltano add file analyze-google-analytics` cannot be built and tested until `models-google-analytics` is published on the HUB

## proposal

The developer creates the plugin definitions and adds using a local file path

NB - is `--from_ref` undocumented https://github.com/meltano/meltano/issues/3208

`meltano add file models-google-analytics --from_ref=../projects/models-google-analytics.yml`
`meltano add file analyze-google-analytics --from_ref=../projects/analyze-google-analytics.yml`

Regular plugin yml definition
`
/projects/analyze-google-analytics.yml
`

Regular plugin yml definition
`
/projects/plugins/models-google-analytics.yml
`

## Related issues

https://github.com/meltano/meltano/issues/3208

"using local hub instance" is not available for local developers https://github.com/meltano/meltano/issues/5997#issuecomment-1146358235

**Comment by tayloramurphy** [comment]
@aphethean1 it seems like adding `--from_ref` would basically solve this problem, right? We'd talked about that in the original spec but it fell out of scope given the time constraints. 

I don't think we'll prioritize this ourselves right now, but I'm in favor of it! cc @aaronsteers @edgarrmondragon

**Comment by edgarrmondragon** [comment]
@tayloramurphy @aphethean1 Note that

> "using local hub instance" is not available for local developers https://github.com/meltano/meltano/issues/5997#issuecomment-1146358235

should no longer the case and developing file bundles with a local Hub instance should be working now: https://github.com/meltano/meltano/pull/5999#issuecomment-1146404199

**Comment by aphethean1** [comment]
@tayloramurphy yes, I think `--from_ref` would be ideal.  Running the hub locally for this development use case seems pretty heavy!

**Comment by WillDaSilva** [comment]
Still relevant.

Side note: If we end up going with `--from_ref` to resolve this issue, it should actually be `--from-ref`. Relates to:
- https://github.com/meltano/meltano/issues/2253
